lib/internal/legacy/processbinding.js

```diff
@@ -34,3 +35,17 @@ module.exports = {
       })));
   }
 };
+
+module.exports = ObjectFromEntries(
+  new SafeArrayIterator(
+    ArrayPrototypeMap(ObjectEntries(factories), ({ 0: key, 1: factory }) => {
+      let result;
+      return [key, () => {
+        if (!result) {
+          result = factory();
+        }
```

Comment:
```suggestion
        result ||= factory();
```
but do we want falsy or nullish?
```suggestion
        result ??= factory();
```

Reply:
Arguably, it can just be `return result ??= factory()`.

Reply:
> Arguably, it can just be `return result ??= factory()`.

I don't know if we have a linter rule enforcing that, but I think we tend to avoid mixing assignments and return statements in the code base, to improve readability.

Reply:
There are a couple of assignments in return statements:
```console
$ git grep "return .* = " lib
```
Since this is a smallish statement, should we prefer `return result ??= factory();`?

Reply:
> > Arguably, it can just be `return result ??= factory()`.
> 
> I don't know if we have a linter rule enforcing that, but I think we tend to avoid mixing assignments and return statements in the code base, to improve readability.

Why would that improve readability?